src/ess/reduce/widgets/__init__.py

```diff
@@ -56,7 +57,23 @@ def wrapper(param: Parameter) -> widgets.Widget:
     return wrapper
 
 
+def optional_widget(
```

Comment:
Is this decorator meant to be used elsewhere? Right now it's only used in one place as far as I can tell.
Maybe we can just put the logic inside that function for now and make it a decorator later if the logic is reused elsewhere?

Reply:
We can't put it inside because of the singledispatch decorator.
You'll have to do it in every dispatch registered functions.

Reply:
Aha I see